src/shogun/lib/View.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Wuwei Lin
+ */
+#ifndef _VIEW__H__
+#define _VIEW__H__
+
+#include <shogun/features/Features.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/lib/SGVector.h>
+#include <type_traits>
+
+namespace shogun
+{
+
+	template <class T>
+	T* view(T* viewable, const SGVector<index_t>& subset)
```

Comment:
is this used anywhere?
How are we going to do this without multiple inheritance?
Maybe a mixin approach would be better, see IterativeMachine

Reply:
@karlnapf i'm not clear, could you explain the multiple inheritance problem?
currently we expect T to be either features or labels, but we can also use viewable mixin in features and labels

Reply:
@karlnapf you mean about the duplciate method or what exactly do you mean by multiple inheritance?

Reply:
I think no.